src/ttlv/format.rs

```diff
@@ -0,0 +1,434 @@
+        if buffer.len() < 1 + num_blocks {
+            return Err(TruncationError);
+        }
+
+        let header = tag.with_type(Type::TextString).value();
+        // NOTE: 'buffer.len() * 8 <= 2^32' so 'num_blocks * 8 < 2^32'.
+        let length = value.len() as u32;
+        let header = (header as u64) << 32 | length as u64;
+        buffer[0].write(header.to_be_bytes());
+
+        // Make sure the last block is completely initialized.
+        if num_blocks > 0 {
+            buffer[num_blocks].write([0; 8]);
+        }
+        let ptr = buffer[1..].as_mut_ptr().cast::<u8>();
+        unsafe { ptr.copy_from_nonoverlapping(value.as_ptr(), value.len()) };
+
+        self.position += 1 + num_blocks;
+        Ok(FormatDone(()))
+    }
+
+    /// Format a byte string.
+    #[inline]
+    pub fn format_bytes(&mut self, tag: Tag, value: &[u8]) -> FormatResult {
+        let num_blocks = value.len().div_ceil(8);
+        let buffer = &mut self.buffer[self.position..];
+        if buffer.len() < 1 + num_blocks {
+            return Err(TruncationError);
+        }
+
+        let header = tag.with_type(Type::ByteString).value();
+        // NOTE: 'buffer.len() * 8 <= 2^32' so 'num_blocks * 8 < 2^32'.
+        let length = value.len() as u32;
+        let header = (header as u64) << 32 | length as u64;
+        buffer[0].write(header.to_be_bytes());
+
+        // Make sure the last block is completely initialized.
+        if num_blocks > 0 {
+            buffer[num_blocks].write([0; 8]);
+        }
+        let ptr = buffer[1..].as_mut_ptr().cast::<u8>();
+        unsafe { ptr.copy_from_nonoverlapping(value.as_ptr(), value.len()) };
+
+        self.position += 1 + num_blocks;
+        Ok(FormatDone(()))
+    }
+
+    /// Format a date-time.
+    #[inline]
+    pub fn format_date_time(&mut self, tag: Tag, value: i64) -> FormatResult {
+        let header = (tag.with_type(Type::DateTime).value() as u64) << 32 | 8;
+
+        let buffer = &mut self.buffer[self.position..];
+        let [header_slot, value_slot, ..] = buffer else {
+            return Err(TruncationError);
+        };
+
+        header_slot.write(header.to_be_bytes());
+        value_slot.write((value as u64).to_be_bytes());
+        self.position += 2;
+        Ok(FormatDone(()))
+    }
+
+    /// Format an interval.
+    #[inline]
+    pub fn format_interval(&mut self, tag: Tag, value: u32) -> FormatResult {
+        let header = (tag.with_type(Type::Interval).value() as u64) << 32 | 4;
+
+        let buffer = &mut self.buffer[self.position..];
+        let [header_slot, value_slot, ..] = buffer else {
+            return Err(TruncationError);
+        };
+
+        header_slot.write(header.to_be_bytes());
+        value_slot.write(((value as u64) << 32).to_be_bytes());
+        self.position += 2;
+        Ok(FormatDone(()))
+    }
+}
+
+//----------- FieldFormatter -------------------------------------------------
+
+/// A [`Formatter`] for filling a particular TTLV field.
```

Comment:
It's a bit strange to link to `Formatter` saying this is one when it is not, because actually it is a `FieldFormatter`?